stylo: Make -moz-{tab-size, box-flex, image-region, outline-radius} animatable.

BorisChiou opened the pull request:
Make -moz-tab-size, -moz-box-flex, -moz-image-region, and -moz-outline-radius animatable, so test_transitions_per_property.html doesn't get any exception of these properties.

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix Bug 1370803, Bug 1370808, Bug 1370845, and Bug 1370846.
- [X] These changes do not require tests because we have test in Gecko side already.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/17223)
<!-- Reviewable:end -->

BorisChiou committed:
Make -moz-tab-size animatable.

BorisChiou committed:
Make -moz-box-flex animatable.

highfive commented:
Heads up! This PR modifies the following files:
 * @bholley: components/style/properties/longhand/inherited_text.mako.rs, components/style/properties/gecko.mako.rs
 * @emilio: components/style/properties/longhand/inherited_text.mako.rs, components/style/properties/gecko.mako.rs

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify style code, but no tests are modified. Please consider adding a test!

BorisChiou commented:
@bors-servo r-hiro

BorisChiou commented:
@bors-servo r=hiro

bors-servo commented:
:pushpin: Commit a75c1dc has been approved by `hiro`

<!-- @bors-servo r=hiro a75c1dc44c21a7670493effa20b92e26f50c155c -->

BorisChiou commented:
@bors-servo r-

I would like to put other patches together.

BorisChiou commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit e0571eedb710ed4d61cbbcc5b4692ca6b9816632 with merge 539e5b4cfe3519bc9b0b0438ca3fad1b36ea0403...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/7390), [arm32](http://build.servo.org/builders/arm32/builds/7604), [arm64](http://build.servo.org/builders/arm64/builds/7643), [linux-dev](http://build.servo.org/builders/linux-dev/builds/7732), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/4488), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/4512), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/7715), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/199), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/198), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/4418), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/4482), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/197), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/200), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/2959)
State: approved= try=True

BorisChiou committed:
Make -moz-image-region animatable.

BorisChiou committed:
Set zero to mImageRegion.{height,width} if {bottom,right} is auto.

Let's follow how Gecko does for this part (in nsRuleNode::ComputeListData).
We set mImageRegion.height or mImageRegion.width to zero if bottom or right is
None (i.e. auto).

BorisChiou committed:
Make -moz-outline-radius-{*} animatable.

BorisChiou commented:
@bors-servo r=hiro,Manishearth

bors-servo commented:
:pushpin: Commit 5c915bd has been approved by `hiro,Manishearth`

<!-- @bors-servo r=hiro,Manishearth 5c915bd4aebf936550ae7b8a3b4ea7a7dcd55b7c -->

bors-servo commented:
:hourglass: Testing commit 5c915bd4aebf936550ae7b8a3b4ea7a7dcd55b7c with merge e386e4248195a17eed522ac8746963022180a6e2...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/7414), [arm32](http://build.servo.org/builders/arm32/builds/7628), [arm64](http://build.servo.org/builders/arm64/builds/7668), [linux-dev](http://build.servo.org/builders/linux-dev/builds/7757), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/4512), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/4536), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/7739), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/224), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/223), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/4442), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/4509), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/221), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/224), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/2984)
Approved by: hiro,Manishearth
Pushing e386e4248195a17eed522ac8746963022180a6e2 to master...
